pkg/conf/schedulerconf.go

```diff
@@ -456,13 +460,43 @@ func DumpConfiguration() {
 	}
 }
 
+func Decompress(key string, value []byte) (string, string) {
+	var uncompressedData string
+	decodedValue := make([]byte, base64.StdEncoding.DecodedLen(len(value)))
+	n, err := base64.StdEncoding.Decode(decodedValue, value)
+	if err != nil {
+		log.Log(log.ShimConfig).Error("failed to decode schedulerConfig entry", zap.Error(err))
+	}
+	decodedValue = decodedValue[:n]
+	splitKey := strings.Split(key, ".")
+	compressionAlgo := splitKey[len(splitKey)-1]
+	if strings.EqualFold(compressionAlgo, constants.GzipSuffix) {
+		reader := bytes.NewReader(decodedValue)
+		gzReader, err := gzip.NewReader(reader)
```

Comment:
gzip reader should be closed to get all errors. There are cases that the reader returns a different error on closing than during processing. Need to handle that as it could surface data corruptions

Form a golang nuts discussion:
https://go.dev/play/p/gwDnxVSQEM4